packages/core/utils/workspace.ts

```diff
@@ -76,7 +78,22 @@ export async function populateWorkspaceDetails(
 
 	const { data: packageJson } = await getPackageJson(workspace);
 	if (packageJson.devDependencies) {
-		workspace.typescript.installed = 'tslib' in packageJson.devDependencies;
+		const tsConfigFileName = 'tsconfig.json';
+		const viteConfigFileName = 'vite.config.ts';
+		let tsConfigExists,
+			viteConfigExists = false;
+
+		if (remoteControl.isRemoteControlled()) {
+			// while executing tests, we only look into the direct `workingDirectory`
+			// as we might detect the monorepo `tsconfig.json` otherwise.
+			tsConfigExists = await fileExists(path.join(workingDirectory, tsConfigFileName));
+			viteConfigExists = await fileExists(path.join(workingDirectory, viteConfigFileName));
+		} else {
+			tsConfigExists = await findUp(workingDirectory, tsConfigFileName);
+			viteConfigExists = await findUp(workingDirectory, viteConfigFileName);
+		}
+
+		workspace.typescript.installed = tsConfigExists || viteConfigExists;
```

Comment:
it might be overkill to also be traversing parent directories for the vite config as well. it should be fine to just check the cwd for the `vite.config.ts` and only traverse up for `tsconfig.json`

```suggestion
		let usesTypescript = false;

		if (remoteControl.isRemoteControlled()) {
			// while executing tests, we only look into the direct `workingDirectory`
			// as we might detect the monorepo `tsconfig.json` otherwise.
			usesTypescript = 
				(await fileExists(path.join(workingDirectory, viteConfigFileName))) ||
				(await fileExists(path.join(workingDirectory, tsConfigFileName)));
		} else {
			usesTypescript = 
				(await fileExists(workingDirectory, viteConfigFileName)) ||
				(await findUp(workingDirectory, tsConfigFileName));
		}
		
		workspace.typescript.installed = usesTypescript;
```